weaviate_agents/personalization/query.py

```
@@ -0,0 +1,189 @@
+        return_references: Optional[ReturnReferences[dict]] = None,
+    ) -> PersonalizedQueryResponse:
+        query_parameters = HybridQueryParameters(
+            query=query,
+            alpha=alpha,
+            vector=vector,
+            query_properties=query_properties,
+            fusion_type=fusion_type,
+            max_vector_distance=max_vector_distance,
+            limit=limit,
+            offset=offset,
+            auto_limit=auto_limit,
+            filters=filters,
+            rerank=rerank,
+            target_vector=target_vector,
+            include_vector=include_vector,
+            return_metadata=return_metadata,
+            return_properties=return_properties,
+            return_references=return_references,
+        )
+        return self._get_query_response(query_parameters)
+
+    def _get_request_data(self, query_parameters: QueryParameters) -> dict[str, Any]:
+        query_request = QueryRequest.model_validate(
+            {
+                "persona_id": self.persona_id,
+                "strength": self.strength,
+                "recent_interactions_count": self.recent_interactions_count,
+                "decay_rate": self.decay_rate,
+                "overfetch_factor": self.overfetch_factor,
+                "query_parameters": query_parameters,
+            }
+        )
+        return {
+            "query_request": query_request.model_dump(mode="json"),
+            "personalization_request": self.personalization_request.model_dump(
+                mode="json"
+            ),
+        }
+
+    def _get_query_response(
+        self, query_parameters: QueryParameters
+    ) -> PersonalizedQueryResponse:
+        response = httpx.post(
+            self._route,
+            headers=self._headers,
+            json=self._get_request_data(query_parameters),
+            timeout=self.timeout,
+        )
+        response_data = response.raise_for_status().json()
```

`raise_for_status` only returns the status code message to the user.
And we often return detailed error messages in response body/
So most flexible way to handle errors is like we do in other places:
```
if response.is_error:
    raise Exception(response.text)
```

---

Oops, I thought these were equivalent, didn't realise that `raise_for_status` swallows the error text - changed 👍

---

I def wish it were 😅